content/docs/guides/auth/ALTS.md

```diff
@@ -0,0 +1,249 @@
+gRPC stack. Google Cloud Platform fully manages the ALTS credentials, including
+certificate issuing, certificate rotation, and certification revocation.
+
+### gRPC Client with ALTS Transport Security Protocol
+
+gRPC clients can use ALTS credentials to connect to servers. The following
+provides examples of how to set up such clients in all supported gRPC languages.
+
+#### C++
+
+```cpp
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/security/credentials.h>
+
+using grpc::experimental::AltsCredentials;
+using grpc_impl::experimental::AltsCredentialsOptions;
+
+auto creds = AltsCredentials(AltsCredentialsOptions());
+std::shared_ptr<grpc::Channel> channel = CreateChannel(server_address, creds);
+```
+
+#### Java
+
+```java
+import io.grpc.alts.AltsChannelBuilder;
+import io.grpc.ManagedChannel;
+
+ManagedChannel managedChannel =
+    AltsChannelBuilder.forTarget(serverAddress).build();
+```
+
+#### Go
+
+```go
+import (
+  "google.golang.org/grpc"
+  "google.golang.org/grpc/credentials/alts"
+)
+
+altsTC := alts.NewClientCreds(alts.DefaultClientOptions())
+conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(altsTC))
+```
+
+#### Python
+
+```python
+import grpc
+
+channel_creds = grpc.alts_channel_credentials()
+channel = grpc.secure_channel(address, channel_creds)
+```
+
+### gRPC Server with ALTS Transport Security Protocol
+
+gRPC servers can use ALTS credentials to allow clients to connect to them. The
+following provides examples of how to set up such servers in all supported gRPC
+languages.
+
+#### C++
+
+```cpp
+#include <grpcpp/security/server_credentials.h>
+#include <grpcpp/server.h>
+#include <grpcpp/server_builder.h>
+
+using grpc::experimental::AltsServerCredentials;
+using grpc_impl::experimental::AltsServerCredentialsOptions;
+
+grpc::ServerBuilder builder;
+builder.RegisterService(&service);
+auto creds = AltsServerCredentials(AltsServerCredentialsOptions());
+builder.AddListeningPort("[::]:<port>", creds);
+std::unique_ptr<Server> server(builder.BuildAndStart());
+```
+
+#### Java
+
+```java
+import io.grpc.alts.AltsServerBuilder;
+import io.grpc.Server;
+
+Server server = AltsServerBuilder.forPort(<port>)
+    .addService(new MyServiceImpl()).build().start();
+
+```
+
+#### Go
+
+```go
+import (
+  "google.golang.org/grpc"
+  "google.golang.org/grpc/credentials/alts"
+)
+
+altsTC := alts.NewClientCreds(alts.DefaultClientOptions())
+conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(altsTC))
+```
+
+#### Python
+
+```python
+import grpc
+
+server = grpc.server(futures.ThreadPoolExecutor())
+server_creds = grpc.alts_server_credentials()
+server.add_secure_port(SERVER_ADDRESS, server_creds)
+```
+
+### Server Authorization
+
+gRPC has built-in server authorization support using ALTS. A gRPC client using
+ALTS can set the expected service accounts prior to establishing a connection.
+Then, at the end of the handshake, it is guaranteed that the server identity
```

Comment:
Just a minor rewording:

```
Then, at the end of the handshake, server authorization guarantees
that the server identity matches one of the service accounts specified
by the client. Otherwise, the connection fails.
```